ui/js/controllers/tcjobactions.js

```diff
@@ -0,0 +1,93 @@
+"use strict";
+
+treeherder.controller('TCJobActionsCtrl', [
+    '$scope', '$http', '$uibModalInstance', 'ThResultSetStore',
+    'ThJobDetailModel', 'thTaskcluster', 'ThTaskclusterErrors',
+    'thNotify', 'job', 'repoName', 'resultsetId', 'actionsRender',
+    function($scope, $http, $uibModalInstance, ThResultSetStore,
+             ThJobDetailModel, thTaskcluster, ThTaskclusterErrors, thNotify,
+             job, repoName, resultsetId, actionsRender) {
+        let originalTask;
+        $scope.input = {};
+
+        $scope.cancel = function() {
+            $uibModalInstance.dismiss('cancel');
+        };
+
+        $scope.updateSelectedAction = function() {
+            $scope.input.jsonPayload = JSON.stringify(jsonSchemaDefaults($scope.input.selectedAction.schema), null, 4);
+        };
+
+        $scope.triggerAction = function() {
+            $scope.triggering = true;
+
+            let tc = thTaskcluster.client();
+
+            let actionTaskId = tc.slugid();
+            let actionTask = actionsRender($scope.input.selectedAction.task, _.defaults({}, {
+                taskGroupId: originalTask.taskGroupId,
+                taskId: job.taskcluster_metadata.task_id,
+                task: originalTask,
+                input: JSON.parse($scope.input.jsonPayload),
+            }, $scope.staticActionVariables));
+
+            let queue = new tc.Queue();
+            queue.createTask(actionTaskId, actionTask).then(function() {
+                $scope.$apply(thNotify.send("Custom action request sent successfully", 'success'));
+                $scope.triggering = false;
+                $uibModalInstance.close('request sent');
+            }, function(e) {
+                $scope.$apply(thNotify.send(ThTaskclusterErrors.format(e), 'danger', true));
+                $scope.triggering = false;
+                $uibModalInstance.close('error');
+            });
+        };
+
+        // prevent closing of dialog while we're triggering
+        $scope.$on('modal.closing', function(event) {
+            if ($scope.triggering) {
+                event.preventDefault();
+            }
+        });
+
+        let decisionTaskId = ThResultSetStore.getGeckoDecisionTaskGUID(repoName, resultsetId);
+        if (decisionTaskId) {
+            let originalTaskId = job.taskcluster_metadata.task_id;
+            $http.get('https://queue.taskcluster.net/v1/task/' + originalTaskId).then(
+                function(response) {
+                    originalTask = response.data;
+                    ThJobDetailModel.getJobDetails({
+                        job_guid: decisionTaskId,
+                        title: 'artifact uploaded',
+                        value: 'actions.json'}).then(function(details) {
```

Comment:
For the record if you have the `taskId` of the decision task you can construct the URL for the actions.json artifact as: 
```js
let queue = new taskcluster.Queue();
let url = queue.buildUrl(queue.getlatestArtifact, taskId, 'public/actions.json');
```